### PR TITLE
LPS-155681 Test Automation Creation - master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-155681

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 1]
```diff
@@ -4755,6 +4755,57 @@ return	Array.from(getWebElement("(//div[@aria-label='${fragmentName}'])[${index}
...
+		if (isSet(label)) {
+			AssertTextEquals(
+				fieldLabel = "${label}",
+				index = "${position}",
+				locator1 = "Fragment#INPUT_LABEL",
+				type = "text-input",
+				value1 = "${label}");
+		}
+
+		if (isSet(helpText)) {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:4772]: showHelpText == true and showHelpText == false

The reason why I didn't use showHelpText == true then else is Poshi has a limitation, Poshi didn't regard condition of else as **showHelpText == false**, Poshi will regard other variable same with **showHelpText != true**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		}
+
+		task ("Then the pages designer can see the custom fields at edit mode") {
+			PageEditor.viewInputText(
+				helpText = "Custom Help Text",
+				label = "Custom Content Field",
+				placeholder = "Custom Placeholder");
+		}
+
+		task ("When the pages designer publishes the page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase:104]: When the page designer navigates to content page at view mode in staging site


			ContentPagesNavigator.openViewContentPage(
				pageName = "Test Page Name",
				siteName = "Test Site Name Staging");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+
+			PageEditor.viewInputText(
+				helpText = "Custom Help Text",
+				label = "Custom Content Field",
+				placeholder = "Custom Placeholder");
+		}
+
+		task ("When the pages designer publishes the site to live") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase:119]: When the page designer navigates to content page at view mode in live site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+			Staging.gotoPublishToLive();
+
+			Staging.publishToLive();
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase:123]: ContentPagesNavigator.openViewContentPage(
				pageName = "Test Page Name",
				siteName = "Test Site Name");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+				hideLabel = "true");
+		}
+
+		task ("Then the pages designer cannot see the hidden fields at edit mode") {
+			PageEditor.viewInputTextNotVisible(
+				helpText = "Custom Help Text",
+				label = "Custom Content Field");
+		}
+
+		task ("When the pages designer publishes the page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase:153]: Same with above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+			ContentPagesNavigator.openViewContentPage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+
+			PageEditor.viewInputTextNotVisible(
+				helpText = "Custom Help Text",
+				label = "Custom Content Field");
+		}
+
+		task ("When the pages designer publishes the site to live") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInputWithStaging.testcase:167]: Same with above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase [hunk 7]
```diff
@@ -45,6 +45,206 @@ definition {
...
+		}
+
+		task ("Then the pages designer can see the custom fields at edit mode") {
+			PageEditor.viewInputText(
+				helpText = "Custom Help Text",
+				label = "Custom Content Field",
+				placeholder = "Custom Placeholder");
+		}
+
+		task ("When the pages designer publishes the page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase:96]: When the page designer navigates to content page at view mode

		ContentPagesNavigator.openViewContentPage(
			pageName = "Test Page Name",
			siteName = "Test Site Name");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase [hunk 8]
```diff
@@ -45,6 +45,206 @@ definition {
...
+				hideLabel = "true");
+		}
+
+		task ("Then the pages designer cannot see the hidden fields at edit mode") {
+			PageEditor.viewInputTextNotVisible(
+				helpText = "Custom Help Text",
+				label = "Custom Content Field");
+		}
+
+		task ("When the pages designer publishes the page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase:128]: When the page designer navigates to content page at view mode

			ContentPagesNavigator.openViewContentPage(
				pageName = "Test Page Name",
				siteName = "Test Site Name");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase [hunk 9]
```diff
@@ -45,6 +45,206 @@ definition {
...
+		}
+
+		task ("Then the pages designer can see the translated fields at edit mode") {
+			PageEditor.viewInputText(
+				helpText = "Texto de ayuda personalizado",
+				label = "Campo de contenido personalizado",
+				placeholder = "Marcador de posición personalizado");
+		}
+
+		task ("When the pages designer publishes the page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase:217]: Same as above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase [hunk 10]
```diff
@@ -151,6 +351,69 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-149725. The pages designer cannot type more than 280 characters in the text input field."
+	@priority = "4"
+	test CannotType281CharactersInTextInputField {
+		property test.name.skip.portal.instance = "TextInput#CannotType281CharactersInTextInputField";
+
+		var type280Characters = "The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters";
+		var type281Characters = "The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characterss";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase:360]: inputText

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase [hunk 11]
```diff
@@ -151,6 +351,69 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-149725. The pages designer cannot type more than 280 characters in the text input field."
+	@priority = "4"
+	test CannotType281CharactersInTextInputField {
+		property test.name.skip.portal.instance = "TextInput#CannotType281CharactersInTextInputField";
+
+		var type280Characters = "The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters is 280.The number of characters";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase:359]: actualText

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase [hunk 12]
```diff
@@ -151,6 +351,69 @@ definition {
...
+		var type281Characters = "The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characterss";
+		var acceptedCharacters = "The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters is 281.The number of characters";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/TextInput.testcase:361]: modifiedText

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind modifying according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro (hunk 1 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:4772]: I see, thanks!

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind modifying according to comments? Thanks

Hi @Tim-Cao, PR updated. Would you have a look at? Would you also wait untill https://issues.liferay.com/browse/LPS-156311 be fixed? I couldn't run the whole test due this bug. Just to verify if I may missed something.

**Comment by ozysouza** [comment]
I need to make more changes due https://issues.liferay.com/browse/LPS-155502

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 13]
```diff
@@ -4755,6 +4755,51 @@ return	Array.from(getWebElement("(//div[@aria-label='${fragmentName}'])[${index}
...
+				locator1 = "Fragment#INPUT_LABEL",
+				type = "text-input",
+				value1 = "${label}");
+		}
+		else if ("${showLabel}" == "false") {
+			AssertElementPresent(
+				fieldLabel = "${label}",
+				index = "${position}",
+				locator1 = "Fragment#INPUT_LABEL_INVISIBLE",
+				type = "text-input");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:4776]: type = ${type}

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 14]
```diff
@@ -4755,6 +4755,51 @@ return	Array.from(getWebElement("(//div[@aria-label='${fragmentName}'])[${index}
...
+		if (!(isSet(position))) {
+			var position = "1";
+		}
+
+		if ("${showLabel}" == "true") {
+			AssertTextEquals(
+				fieldLabel = "${label}",
+				index = "${position}",
+				locator1 = "Fragment#INPUT_LABEL",
+				type = "text-input",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:4768]: type = ${type}

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 15]
```diff
@@ -4755,6 +4755,51 @@ return	Array.from(getWebElement("(//div[@aria-label='${fragmentName}'])[${index}
...
+				index = "${position}",
+				locator1 = "Fragment#INPUT_LABEL_INVISIBLE",
+				type = "text-input");
+		}
+
+		if ("${showHelpText}" == "true") {
+			AssertTextEquals(
+				index = "${position}",
+				locator1 = "Fragment#INPUT_HELP_TEXT",
+				type = "input",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:4783]: type = ${type}

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 16]
```diff
@@ -4755,6 +4755,51 @@ return	Array.from(getWebElement("(//div[@aria-label='${fragmentName}'])[${index}
...
+				locator1 = "Fragment#INPUT_HELP_TEXT",
+				type = "input");
+		}
+
+		if (isSet(placeholder)) {
+			AssertAttributeValue(
+				attribute1 = "placeholder",
+				index = "${position}",
+				locator1 = "Fragment#INPUT_FIELD",
+				type = "${placeholderType}",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:4798]: type = ${type}

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating according to comments? Thanks

Hi @Tim-Cao, PR updated. Let's just wait for [LPS-156311](https://issues.liferay.com/browse/LPS-156311).

**Comment by Tim-Cao** [comment]
LGTM! We still needs to case to cover other points.
See https://github.com/brianchandotcom/liferay-portal/pull/119106

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**